src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -255,6 +265,42 @@ DQMatMulGQi::DQMatMulGQi(Context::Ref ctx) {
             act_shape.size() == 3 && act_shape[1] == 1 &&  // single-token case
             qcoeff_shape[0] == qweight_shape[0] && qcoeff_shape[1] == 1 && qcoeff_shape[2] == qweight_shape[2] &&
             !matched_matmul->get_transpose_a() && !matched_matmul->get_transpose_b()) {
+            if (!ctx.get().mm_dq_full) {
+                // Transpose weight and coeff
+                ov::Shape tw_shape = {qweight_shape[0], qweight_shape[2], qweight_shape[1]};
+                matched_qweight->set_partial_shape(tw_shape);
+                matched_qweight->validate_and_infer_types();
+                ctx.get().permute(matched_qweight, {0, 2, 1});
```

Comment:
An idea to refactoring:

Since `.permute()` doesn't create a new Parameter but modifies the closure for the existing one, it could also update the shape as you do with the previous three lines?

```
                ov::Shape tw_shape = {qweight_shape[0], qweight_shape[2], qweight_shape[1]};
                matched_qweight->set_partial_shape(tw_shape);
                matched_qweight->validate_and_infer_types();
```

Reply:
Done